nck/readers/search_console_reader.py

```diff
@@ -65,15 +71,17 @@ def __init__(
         end_date,
         date_column,
         row_limit,
+        date_range,
     ):
         self.client_id = client_id
         self.client_secret = client_secret
         self.access_token = access_token
         self.refresh_token = refresh_token
         self.dimensions = list(dimensions)
         self.site_url = site_url
-        self.start_date = datetime.strftime(start_date, DATEFORMAT)
-        self.end_date = datetime.strftime(self.check_end_date(end_date), DATEFORMAT)
+        self.start_date, self.end_date = build_date_range(start_date, end_date, date_range)
+        self.start_date = datetime.strftime(self.start_date, DATEFORMAT)
+        self.end_date = datetime.strftime(self.check_end_date(self.end_date), DATEFORMAT)
```

Comment:
I didn't change these lines I just adapted them to use "self" instead of the arguments, but I can change the code if you want to keep them as datetime

Reply:
Ok, it's up to you. In my opinion, manipulating native date objects until we **really** need to convert them makes a lot of things easier, especially when we have to make operations on them. Here, we don't do much with the dates, so the way it is implemented is acceptable. This is why it was a "nitpick" 🙂

Reply:
I agree and I changed it to make it cleaner